ptychonn/position/configs.py

```diff
@@ -0,0 +1,354 @@
+    would then require a Reconstructor object that generate reconstructed images from diffraction patterns.
+    Alternatively, one could also keep this argument as None and pass a VirtualReconstructor set to read
+    the reconstructed images to ptycho_reconstructor.
+    """
+
+    dp_data_file_path: str = None
+
+    dp_data_file_handle: Any = None
+    """Used as an alternative to `dp_data_file_path`. Should be a `DataFileHandle` object."""
+
+    probe_position_list: Any = None
+    """
+    A ProbePositionList object used for finding nearest neighbors in collective mode.
+    If None, `probe_position_data_path` must be provided.
+    """
+
+    probe_position_data_path: Any = None
+
+    probe_position_data_unit: str = None
+    """Unit of provided probe position. Can be 'nm', 'm', or 'pixel'."""
+
+    pixel_size_nm: float = None
+
+    baseline_position_list: Any = None
+    """Baseline positions. Used by ProbePositionCorrectorChain when the serial mode result is bad."""
+
+    central_crop: Any = None
+    """
+    List or tuple of int. Patch size used for image registration. If smaller than the reconstructed object size,
+    a patch will be cropped from the center.
+    """
+
+    method: str = "collective"
+    """Method for correction. Can be 'serial' or 'collective'"""
+
+    num_neighbors_collective: int = 3
+    """Number of neighbors in collective registration"""
+
+    offset_estimator_order: int = 1
+
+    offset_estimator_beta: float = 0.5
+
+    smooth_constraint_weight: float = 1e-2
+
+    rectangular_grid: bool = False
+
+    stitching_downsampling: int = 1
+
+    random_seed: Any = 123
+
+    debug: bool = False
+
+
+class TrainingConfigDict(ConfigDict):
+    batch_size_per_process: int = 64
+
+    num_epochs: int = 60
+
+    learning_rate_per_process: float = 1e-3
+
+    optimizer: str = "adam"
+    """String of optimizer name or the handle of a subclass of torch.optim.Optimizer"""
+
+    model_save_dir: str = "."
+    """Directory to save trained models"""
+
+    model: Any = None
+    """
+    The model. The three options are:
+    (1) None: the model will be instantiated with the default model class.
+    (2) A object of nn.Module: the model object will be used as provided.
+
+
+    (3) tuple(nn.Module, kwargs): the first element of the tuple is the class handle of a model class, and the
+        second is a dictionary of keyword arguments. The model will be instantiated using these.
+    """
+
+    l1_weight: float = 0
+
+    tv_weight: float = 0
+
+
+class PtychoNNTrainingConfigDict(TrainingConfigDict):
+    height: int = 256
+
+    width: int = 256
+
+    num_lines_for_training: int = 100
+    """Number of lines used for training"""
+
+    num_lines_for_testing: int = 60
+    """Number of lines used for testing"""
+
+    num_lines_for_validation: int = 805
+    """Number of lines used for testing"""
+
+    dataset: Any = None
```

Comment:
How can the dataset be optional? Maybe you want to move this parameter to the top and make it required by not providing a default value?

Reply:
No parameters should have a type hint of `Any` unless they can truly be anything. Parameters that can be `int` or `None`, for example, should either be `typing.Optional[int]` or `int | None`. The `|` syntax requires a minimum python version of 3.10.

Reply:
Replaced `Any`s with more specific type hints in fdc7ae7.